### DISCUSSION TITLE
Pagination

### DISCUSSION THREAD

**Discussion by david-welch** [BeanieODM/beanie#392] [Feature Request]
Hello,
I was wondering if there are any plans to add pagination support to Beanie, particularly cursor-based? If not, is there a recommended library to use? 

Thanks!

**Comment by syedhabib53** [comment]
There is a [fastapi_pagination](https://github.com/uriyyo/fastapi-pagination) lib, that could help you. Unfortunately support for `beanie` not yet released, since it is in master branch. They planned to release it with their next release, Check this [here](https://github.com/uriyyo/fastapi-pagination/issues/334)

**Reply by david-welch** [reply]
This is very promising, though it doesn't currently support cursor pagination for beanie:
https://github.com/uriyyo/fastapi-pagination/blob/main/fastapi_pagination/ext/beanie.py#L17

**Reply by syedhabib53** [reply]
Any specific reason for cursor based pagination. Considering it can already achieve using `limit` and `offset`. 🙄

**Reply by david-welch** [reply]
As I understand it, cursor has better handling of inserts and deletes. It scales better with data set size as serving pages no longer requires looping through every record, eg: https://slack.engineering/evolving-api-pagination-at-slack/

**Reply by janas-adam** [reply]
hi, came across this old discussion

if u lookin for fastapi-pagination beanie extension example ->

```
from fastapi_pagination.ext.beanie import apaginate
from fastapi_pagination import Page, Params

@users_router.get("/users", response_model=Page[UserReadOnlySchema])
async def list_users(
    ...
):
    return await apaginate(User.find_all(), Params(page=1, size=10))
```

hope it ll help

@07pepa @staticxterm imo this request can be closed
